src/mongocxx/test/collection.cpp

```diff
@@ -2808,4 +2812,124 @@ TEST_CASE("Ensure that the WriteConcernError 'errInfo' object is propagated", "[
     REQUIRE(contains_err_info);
 }
 
+//JFW: TEST_CASE("Prose test 2. WriteError.details exposes writeErrors[].errInfo") {
+//JFW: TEST_CASE("expose writeErrors[].errInfo", "[collection]") {
+TEST_CASE("JFWstuff", "[collection]") {
+
+    instance::current();
+
+    auto client_opts = mongocxx::options::client();
+
+    // Listen to the insertion-failed event: we want to get a copy of the server's
+    // response so that we can compare it to the thrown exception later:
+    {
+    mongocxx::options::apm apm_opts;
+    apm_opts.on_command_failed([](const events::command_failed_event& ev) {
+        std::cout << "<- command failed " << ev.command_name() << " " << bsoncxx::to_json (ev.failure()) << std::endl;
+    });
+    apm_opts.on_command_succeeded ([](const mongocxx::events::command_succeeded_event& ev) {
+
+//        std::cout << "<- command succeeded " << ev.command_name() << " " << bsoncxx::to_json (ev.reply()) << std::endl;
+/* JFW: we'll get something like this back:
+{ "n" : 0, "writeErrors" : [ { "index" : 0, "code" : 121, "errInfo" : { "failingDocumentId" : { "$oid" : "6217bd907048698de70e05f1" }, "details" : { "operatorName" : "$type", "specifiedAs" : { "field_x" : { "$type" : "string" } }, "reason" : "type did not match", "consideredValue" : 42, "consideredType" : "int" } }, "errmsg" : "Document failed validation" } ], "nModified" : 0, "ok" : 1.0 }
+
+...we want to be sure that we got "errInfo" and inside of that a "details" document.
+
+          auto error = e.raw_server_error()->view();
+          auto result = error["writeConcernErrors"][0]["errInfo"];
+          contains_err_info = (err_info == result.get_document().view());
+
+*
+*/
+
+        auto reply_view = ev.reply();
```

Comment:
To check for the presence of a field, this is sufficient:

```cpp
if (reply_view["writeErrors"]) {
    // writeErrors field exists.
}

if (reply_view["writeErrors"][0]["errInfo"]["details"]) {
    // writeErrors[0].errInfo.details exists.
}

// To get the value of a field as a document.
auto element = reply_view["writeErrors"][0]["errInfo"]["details"];
if (element && element.type() == bsoncxx::type::k_document) {
    auto document = element.get_document ();
    std::cout << "details document is " << bsoncxx::to_json (document.view()) << std::endl;
}
```

Reply:
+1 super helpful, thank you!